### ISSUE TITLE
Testing first alpha version

### ISSUE THREAD

**Issue by alfonsogarciacaro** [fable-compiler/fable-scripts#1]
Wow, this was more complicated than I expected but at least I got something. @whitetigle @MangelMaxime Would you mind giving a try to the first version of the fable-scripts? You need to do the following:

```
git clone https://github.com/fable-compiler/fable-scripts
cd fable-scripts/sample
yarn
yarn start
```
Reference: https://github.com/fable-compiler/Fable/issues/1281

I wanted to keep the webpack.config.js file hidden but I got many problems with package references, so now I'm just copying it to the root of the sample app. Maybe we could put it in a hidden directory (or maybe problems happened because I tested it when the original webpack.config.js was in a directory above the sample app, and they will solve when the sample app is installed inside sample's `node_modules` 🤔).

Rigth now, `yarn start` calls `fable-scripts start`. The idea is to do something similar with `build`, `test`... so we can change the script behaviour with new versions of fable-scripts without touching the package.json. Something I don't like very much is we're losing the colors in the console output, I'm not sure how to fix that.

Please let me know if it works for you and your thoughts.

**Comment by whitetigle** [comment]
Yes captain! Currently cloning the repo!

**Comment by whitetigle** [comment]
It fails on my side after calling `yarn start`

```
fable-scripts start
CWD: D:\workspace\fable-scripts\sample\src
C:\Program Files\dotnet\dotnet.EXE fable webpack-dev-server --port free
events.js:160
      throw er; // Unhandled 'error' event
      ^

Error: spawn C:\Program ENOENT
    at exports._errnoException (util.js:1022:11)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:193:32)
    at onErrorNT (internal/child_process.js:359:16)
    at _combinedTickCallback (internal/process/next_tick.js:74:11)
    at process._tickCallback (internal/process/next_tick.js:98:9)
    at Module.runMain (module.js:607:11)
    at run (bootstrap_node.js:420:7)
    at startup (bootstrap_node.js:139:9)
    at bootstrap_node.js:535:3
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by whitetigle** [comment]
```console
λ dotnet fable yarn-splitter -- -w
Fable (1.3.2) daemon started on port 61225
CWD: D:\workspace\fable-scripts\sample
cmd /C yarn run splitter -w
yarn run v1.3.2
warning You are using Node "7.4.0" which is not supported and may encounter bugs or unexpected behavior. Yarn supports the following semver range: "^4.8.0 || ^5.7.0 || ^6.2.2 || >=8.0.0"
error Command "splitter" not found.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
Closing Fable daemon...
```

**Comment by whitetigle** [comment]
The command fable-splitter does not seem to work. How do you call it?

**Comment by alfonsogarciacaro** [comment]
Ah, damn! I need to wrap the command in quotes. I updated the version, can you please try again?

```
git pull
cd sample
yarn
yarn start
```

If you want to build fable-scripts you have to also run `yarn` in repo root:

```
yarn
yarn build
```

**Comment by whitetigle** [comment]
Just for the log here. It's working on my side.
👏 👏 👏 🍾 🍾 🍾 🍾

**Comment by alfonsogarciacaro** [comment]
And in latest version now webpack.config is hidden and most of the dependencies have been removed from the sample app package.json 💃

**Comment by MangelMaxime** [comment]
I didn't test it yet, but I don't understand the goal of this project. Could you please explain it :) ?

I know it have something to do with the issues that you linked. But what usage does it cover ? Is it some kind of a template ? Something that will update the project of the user or just kick start it ?

**Comment by alfonsogarciacaro** [comment]
@MangelMaxime Right now we need configuration for several JS tools in Fable projects, like:

- webpack.config.js
- Babel (usually integrated in webpack.config)
- Package.json scripts

At the moment we use templates to establish a recommendation for the configuration and make it easier for developers to get started. But when we want to change something in the recommendation for the configuration then there are two problems:

- We have to make the change in all templates and it's difficult to keep them in sync. For example, I changed Fable.Template to use babel-env, but Elmish templates are still using babel-presets.
- For users it's very complicated to update existing projects, because they have to download the new template, extract the config files, compare them with their current ones, etc (copy & paste is not OK because they may overwrite custom changes).

This project tries to maintain a standard set of configuration files, [in a similar way as react-scripts do](https://github.com/facebookincubator/create-react-app/tree/master/packages/react-scripts), that can be easily shared among templates/projects and it's easy to update too (you only need to update fable-scripts without touching any file). We also bring back fableconfig.json, so users only need to configure a few options (and they can have intellisense with Ionide). As a bonus, it also removes many of the development packages from package.json (as they're now dependencies of fable-scripts).

Power-users can of course keep handling their own Webpack configuration :+1:

**Comment by MangelMaxime** [comment]
Hum, ok I think I see the goal. 

And how would they include fable-script into their project ? Via git or using an npm/nuget package ?

**Comment by alfonsogarciacaro** [comment]
Npm package should be enough. React uses create-react-app which adds a react-scripts dependency to the new project. But we already have dotnet templates so we can add the fable-scripts dependency there (and remove most of the current ones: fable-loader, webpack, etc).

**Comment by whitetigle** [comment]
Dumb question: when can we start using it? (sooooooo impatient)

**Comment by MangelMaxime** [comment]
I just tested it with the commands you listed @alfonsogarciacaro and it's working.

Tested under Linux

**Comment by alfonsogarciacaro** [comment]
@whitetigle I just tested the scripts with my project at work and they're working. I was thinking to publish it with a new template with Elmish+Fulma combo (and maybe also server side) but I can give you a preview ;) Gimme a couple of days.

**Comment by whitetigle** [comment]
Don't you worry @alfonsogarciacaro. No hurry there. Take your time. I have lots of things to do myself. It was just a way for me to express my real enthusiasm for this new project. 😁

**Comment by alfonsogarciacaro** [comment]
It seems Webpack 4 will have some defaults for input and output directories. It's probably a good to follow them:

https://twitter.com/jdalton/status/951250082791227392

**Comment by nojaf** [comment]
I'm guessing this first version is a no go if you don't have `yarn` installed?
https://github.com/fable-compiler/fable-scripts/blob/master/package.json#L13

**Comment by alfonsogarciacaro** [comment]
Yes, atm you need `yarn`, though I guess we could change the script so it doesn't yarn (off the top of my head, something like `"start": "cd src && dotnet fable node-run ../node_modules/fable-splitter/cli.js -- -c ../splitter.config.js -w"`).

Anyways, we're exploring other possibilities atm (Webpack 4, Parcel) so I haven't released a beta version of this yet. But please feel free to try it out and give your feedback :)
